app/src/main/java/org/piwigo/ui/main/AlbumsFragment.java

```diff
@@ -82,4 +86,27 @@ private int calculateColumnCount() {
         boolean largeScreen = screenSize == Configuration.SCREENLAYOUT_SIZE_LARGE || screenSize == Configuration.SCREENLAYOUT_SIZE_XLARGE;
         return (int) Math.floor(configuration.screenWidthDp / (largeScreen ? TABLET_MIN_WIDTH : PHONE_MIN_WIDTH));
     }
+
+    public void refresh() {
+        /* TODO: implement real refresh and call setRefreshing(false) after finishing the refresh on the swipe (see onRefresh)
+        * maybe the viewModel.loadAlbums(null) already does the job? but is this synchronous and fully refreshed on return?
+        * */
+        viewModel.loadAlbums(null);
+    }
+
+    @Override
+    public void onRefresh(){
+
+        Toast.makeText(getActivity().getApplicationContext(), "Refreshing is not yet implemented :-(", Toast.LENGTH_LONG).show();
```

Comment:
At this early stage I think at most a TODO with a corresponding issue id would suffice

Reply:
clearly yes. This was one of the first pieces of code in the app where I was adding something "visible", so it made me happy.
On the other side, the UI showed the "update" animation, suggesting to the user that there is something ongoing, but in reality nothing happens, that's why I finally decided to commit it because who knows when this will be implemented - I think there should be a release before we care about that, and removing the SwipeRefreshLayout until its implemented is also undo of a work that finally will be good.